apiserver/charms.go

```diff
@@ -444,24 +444,31 @@ type CharmUploader interface {
 // RepackageAndUploadCharm expands the given charm archive to a
 // temporary directory, repackages it with the given curl's revision,
 // then uploads it to storage, and finally updates the state.
-func RepackageAndUploadCharm(ctx context.Context, objectStore services.Storage, uploader CharmUploader, archive *charm.CharmArchive, curl string, charmRevision int) error {
+func RepackageAndUploadCharm(
+	ctx context.Context,
+	objectStore services.Storage,
+	uploader CharmUploader,
+	archive *charm.CharmArchive,
+	curl string,
+	charmRevision int,
+) (charm.Charm, string, string, string, error) {
```

Comment:
Comment should be updated to reflect these new returns, even if it's scaffolding, with a note regarding later removal if required.